### PR TITLE
fix: embed z awk script to __z*.fish files; close #66

### LINKED ISSUE #66: Doesn't work with fisherman v3
For reference https://github.com/jorgebucaran/fisher/issues/452
@jorgebucaran 

```
> fisher add jethrokuan/z
fetching https://codeload.github.com/edc/bass/tar.gz/master
fetching https://codeload.github.com/jethrokuan/z/tar.gz/master
linking ~/.config/fish/functions/bass.fish
linking ~/.config/fish/functions/__z.fish
linking ~/.config/fish/functions/__z_add.fish
linking ~/.config/fish/functions/__z_clean.fish
linking ~/.config/fish/functions/__z_complete.fish
linking ~/.config/fish/conf.d/z.fish
added 1 package, updated 1 package in 0.69s
> z
awk: can't open file /Users/arkadyan/.config/fish/functions/z.awk
 source line number 1 source file /Users/arkadyan/.config/fish/functions/z.awk
 context is
	 >>>  <<< 
```

### PR DESCRIPTION
**Pull request by jorgebucaran**
This PR embed z's awk script to __z*.fish files, fixing #66 and restoring fisher support.

@jethrokuan I also took the liberty to remove the unused fishfile and add some minor improvements to the README file. 🙏

### CODE REVIEW

File: functions/__z_clean.fish [hunk 1]
```diff
@@ -1,9 +1,8 @@
 function __z_clean -d "Clean up .z file to remove paths no longer valid"
-  set -l path (command dirname (status -f))
   set -l tmpfile (mktemp $Z_DATA.XXXXXX)
 
   if test -f $tmpfile
-    command awk -F "|" -f $path/zclean.awk $Z_DATA > $tmpfile
+    command awk -F "|" 'system("test -d \"" $1 "\"") == 0 { print $0 }' $Z_DATA > $tmpfile
```
Comment (jorgebucaran) [inline functions/__z_clean.fish]: @jethrokuan I think this could be simplified a bit, but I'll leave it for a futuer PR.

**Comment by amir20** [comment]
It works!

**Comment by jorgebucaran** [comment]
🎉

**Comment by jethrokuan** [comment]
Thanks!

**Comment by jorgebucaran** [comment]
👋 Thanks @jethrokuan!

If anyone is still using my fork, please edit your fishfile to replace it with jethrokuan's Z.

```
code ~/.config/fish/fishfile
```

```diff
- jorgebucaran/z
+ jethrokuan/z
```

...and run `fisher` to commit your changes!

```
fisher
```

Cheers!

cc @amir20 @lardissone @arkadyan @MaxMilton @himkt

**Pull request merged**